### PR TITLE
Add: Validate personal and ID numbers on altinn send

### LINKED ISSUE #182: Validate ID number when sending to altinn
2016-06-13_13-54-37_altinn_personal_id

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #182

### CODE REVIEW

File: modules/altinnsalary/model/altinnreport.class.php [hunk 1]
```diff
@@ -247,10 +247,12 @@ function generateInntektsmottakerArray($key_subcompany, $salary, $employee, $wor
...
-    $society_number_is_empty = self::checkIfEmpty($society_number, 'Ansatt: Mangler personnummer for ' . self::fullNameForErrorMessage($employee));
+    $society_and_id_number_are_empty = self::checkIfEmpty($society_number.$id_number, 'Ansatt: Mangler personnummer og ID nummer for ' . self::fullNameForErrorMessage($employee));
     // Error is: Personal ID must be valid by mod11
-    if(!$society_number_is_empty) self::checkIfEmpty($society_number, "Ansatt: Personnr m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
+    if(!$society_and_id_number_are_empty && !empty($society_number)) self::checkIfEmpty($society_number, "Ansatt: Personnr m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
+    else if(!$society_and_id_number_are_empty && !empty($id_number)) self::checkIfEmpty($id_number, "Ansatt: ID nummer m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
 
     $inntektsmottaker['inntektsmottaker']['norskIdentifikator'] = $society_number;
```
Comment (Baxxx) [inline modules/altinnsalary/model/altinnreport.class.php]: In this case we always use 

``` php
$inntektsmottaker['inntektsmottaker']['norskIdentifikator'] = $society_number;
```

@stoivo @mladjo2505 When we send this report should we use $id_number as 'norskIdentifikator'? Does Altinn work that way?

Comment (mladjo2505) [inline modules/altinnsalary/model/altinnreport.class.php]: @stoivo We should probably ask Arnt.

Comment (stoivo) [inline modules/altinnsalary/model/altinnreport.class.php]: Yes we should use $id_number as norskIdentifikator if $society_number is blank.

**Comment by mladjo2505** [comment]
Went through code with @MrdjanStojimirovic 
Final review with @stoivo

File: modules/altinnsalary/model/altinnreport.class.php [hunk 2]
```diff
@@ -247,12 +247,14 @@ function generateInntektsmottakerArray($key_subcompany, $salary, $employee, $wor
...
+    $society_and_id_number_are_empty = self::checkIfEmpty($society_number.$id_number, 'Ansatt: Mangler personnummer og ID nummer for ' . self::fullNameForErrorMessage($employee));
     // Error is: Personal ID must be valid by mod11
-    if(!$society_number_is_empty) self::checkIfEmpty($society_number, "Ansatt: Personnr m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
+    if(!$society_and_id_number_are_empty && !empty($society_number)) self::checkIfEmpty($society_number, "Ansatt: Personnr m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
+    else if(!$society_and_id_number_are_empty && !empty($id_number)) self::checkIfEmpty($id_number, "Ansatt: ID nummer m&aring; v&aelig;re gyldig mod11 for " . self::fullNameForErrorMessage($employee), 'personal_number');
 
-    $inntektsmottaker['inntektsmottaker']['norskIdentifikator'] = $society_number;
+    $inntektsmottaker['inntektsmottaker']['norskIdentifikator'] = $society_number != "" ? $society_number : $id_number;
```
Comment (Baxxx) [inline modules/altinnsalary/model/altinnreport.class.php]: Use empty() instead of != ""

**Pull request merged**